### DISCUSSION TITLE
Internal database or PostgreSQL for a new budibase application

### DISCUSSION THREAD

**Discussion by harriejean** [Budibase/budibase#9506] [Help] [answered]
Hi,

I have a basic question regarding Budibase and PostgreSQ. I am in the process of developing a booking system for a small yoga studio. At the moment there is no data at all.

Does it make sense to use an external database (e.g. PostgreSQL) with a sophisticated data structure in this context, if Budibase (selfhosted at DigitalOcean) does not import the corresponding relationships and constraints at all? Or does it?

Would I have advantages using an external PostgreSQL database?
Or would Budibase's internal database be sufficient?

Thanks in advance to all...

Best regards

harriejean

**Comment by melohagan** [comment, marked as answer]
Hey @harriejean 

I would expect for a simple booking system the internal Budibase DB would do the job. I can't imagine you would have too many relationships between your tables?

The video here could be worth watching as well! https://docs.budibase.com/docs/relationships#considerations

**Reply by harriejean** [reply]
Hey melohagan,

Thank you for your answer.

It is often the case that things start small and then expand "unexpectedly". A redesign is then often omitted in view of the effort involved, although it would certainly make sense for reasons of security.

I will have a look at the video you suggested.

For your information, here is the table model of the external PostgreSQL database created with DBeaver:

![image](https://user-images.githubusercontent.com/12449454/216452806-5141fd0a-02c2-45cc-9d23-026ac2982116.jpeg)

It will certainly be extended. In addition, I wanted to keep the possibility of possibly migrating to another application platform. That's why I came up with the idea of a separate database from the application.

Best regards and thanks again.

HarrieJean

Update:
After watching the video, it is completely clear to me that I will continue to use PostgreSQL as an external database. If only because of the many query possibilities.

Thx again, melohagan.
